edk2toolext/environment/repo_resolver.py

```diff
@@ -598,15 +600,18 @@ def submodule_resolve(
         repo.git.submodule("sync", "--", submodule.path)
 
         params = ["update", "--init"]
-        if submodule.recursive:
+        if submodule.recursive and not submodule.configuration_file:
```

Comment:
I would argue that `submodule.recursive` and `submodule.configuration_file` both being set by the consumer should be a error and exit or at least a warning, rather than quietly ignoring the recursive flag.

Any reason I'm thinking wrong with that?

Reply:
I was trying to user proof it.

If the user was making a change and missed changing recursive to false but provided a config file, I was defering to config file.

There are a lot more cases of users using True today then using the config file....

Reply:
haha fair enough. :)